### PR TITLE
Use the RKNG

### LINKED ISSUE #1909: Implement proper integration for flight plans
We do things with 100 small steps which is quite ridiculous, see:
https://github.com/mockingbirdnest/Principia/blob/e3f9ad66303538deb9d02a6825c62bf04b09bcb8/ksp_plugin/flight_plan.cpp#L326

### PR DESCRIPTION
**Pull request by eggrobin**
Fix #1909.

### CODE REVIEW

File: ksp_plugin/flight_plan.cpp [hunk 1]
```diff
@@ -321,38 +323,35 @@ void FlightPlan::BurnLastSegment(NavigationManœuvre const& manœuvre) {
...
-                                             movement.IntrinsicAcceleration(),
-                                             movement.final_time(),
-                                             adaptive_step_parameters_,
-                                             max_ephemeris_steps_per_frame,
-                                             /*last_point_only=*/false).ok();
-        if (!reached_desired_final_time) {
-          anomalous_segments_ = 1;
-          return;
-        }
+      // TODO(egg): move this to the Manœuvre class.
```
Comment (pleroy) [inline ksp_plugin/flight_plan.cpp]: Please create an issue for all these TODOs.

File: physics/ephemeris.hpp [hunk 2]
```diff
@@ -54,6 +55,11 @@ class Ephemeris {
   using IntrinsicAcceleration =
       std::function<Vector<Acceleration, Frame>(Instant const& time)>;
```
Comment (pleroy) [inline physics/ephemeris.hpp]: Do we still need the old intrinsic acceleration somewhere?  If not, mention that we need to clean up in the issue.

File: physics/ephemeris_body.hpp [hunk 3]
```diff
@@ -552,6 +555,115 @@ Status Ephemeris<Frame>::FlowWithAdaptiveStep(
...
+  // The |min| is here to prevent us from spending too much time computing the
+  // ephemeris.  The |max| is here to ensure that we always try to integrate
+  // forward.  We use |last_state_.time.value| because this is always finite,
+  // contrary to |t_max()|, which is -∞ when |empty()|.
+  Instant const t_final =
+      std::min(std::max(instance_time() +
+                            max_ephemeris_steps * parameters_.step(),
+                        trajectory_last_time + parameters_.step()),
+               t);
+  Prolong(t_final);
```
Comment (pleroy) [inline physics/ephemeris_body.hpp]: I am sure that I have seen most of this code elsewhere.  Please factor it out.

File: physics/ephemeris.hpp (hunk 2 above)
Comment (eggrobin) [reply physics/ephemeris.hpp]: We still use it for the non-guided path, which uses a non-RKNG RKN.

File: physics/ephemeris_body.hpp (hunk 3 above)
Comment (eggrobin) [reply physics/ephemeris_body.hpp]: More TODOs for #1931.

File: ksp_plugin/flight_plan.cpp (hunk 1 above)
Comment (eggrobin) [reply ksp_plugin/flight_plan.cpp]: #1931.

**Pull request merged**